### PR TITLE
Infra/lodash babel plugin

### PR DESCRIPTION
**Pull request by lidord-wix**
## Description
Reduce bundle size using lodash babel plugin - https://github.com/lodash/babel-plugin-lodash 
As part of it, I was needed to migrate all the usages of `_.chain` to `_.flow`, since `_.chain` brings all lodash with it. 
**I recommend reviewing this PR by commit since one of the commits is only for formatting**
WOAUILIB-2485

## Changelog
Reduce bundle size using lodash babel plugin

### CODE REVIEW

**Review by M-i-k-e-l — COMMENTED** [review]
A general comment, I see a lot of places where you wrote `arr` (mainly) or `str` and not used the original name, was that intentional or some autogeneration?

**Comment by lidord-wix** [comment]
> A general comment, I see a lot of places where you wrote `arr` (mainly) or `str` and not used the original name, was that intentional or some autogeneration?

Actually, I preferred to use `arr | str | obj` as a parameter to make it clear that it's a pure function that I pass to `_.flow()`,
but if it's not clear I can change it.
I can also save those functions in a different consts and pass them that way if you find it more readable

**Comment by M-i-k-e-l** [comment]
> Actually, I preferred to use `arr | str | obj` as a parameter to make it clear that it's a pure function that I pass to `_.flow()`, but if it's not clear I can change it.

I think that if you have a proper name (`items`) you should use that, in other places (`keys` of `props`) it might be ok to use `arr`.
This might be a matter of taste, so maybe we need anther dev's input on this, your choice.

> I can also save those functions in a different consts and pass them that way if you find it more readable

Tough one, it might be more readable but it might be less readable with wrong name(s), it will also make the code a bit longer, I think this is generally better (unless you have really good names and \ or repeating functions in the same class).

File: src/assets/Assets.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
   if (a) {
-    _(b).keys().forEach((key) => {
-      // @ts-ignore
-      Object.defineProperty(a, key, Object.getOwnPropertyDescriptor(b, key));
-    });
+    _.flow(_.keys, arr =>
+      _.forEach(arr, key => {
+        // @ts-ignore
+        Object.defineProperty(a, key, Object.getOwnPropertyDescriptor(b, key));
+      }))(b);
```
Comment (ethanshar) [inline src/assets/Assets.ts:13]: Why did you change here if there's no _.chain?

**Review by ethanshar — COMMENTED** [review]
Just FYI, there was no reason to change usages outside of the `src` files (: 
We only deliver the src files and we babel only them. 

Anyway, Have you checked If the exported files got their lodash imports changed?

File: src/assets/Assets.ts (hunk 1 above)
Comment (lidord-wix) [reply src/assets/Assets.ts:13]: It's an unexplicit chain - https://lodash.com/docs/4.17.15#prototype-chain

**Comment by lidord-wix** [comment]
> Just FYI, there was no reason to change usages outside of the `src` files (: We only deliver the src files and we babel only them.
> 
> Anyway, Have you checked If the exported files got their lodash imports changed?

Yes, it does change the imports :)

File: src/assets/Assets.ts (hunk 1 above)
Comment (ethanshar) [reply src/assets/Assets.ts:13]: Wouldn't it be enough to just change it to `_.keys(b).forEach`? Would that work? 
I personally find the _.flow not very readable, so whenever possible, I'd use a different approach. 

In this case we can also use vanilla JS and do `Object.keys(b).forEach`

Comment (lidord-wix) [reply src/assets/Assets.ts]: Done.
I changed to vanilla JS where I found it more readable and also renamed the parameters of the methods as Miki suggested.

**Review by ethanshar — APPROVED** [review]
